editor/create_dialog.cpp

```diff
@@ -122,6 +117,29 @@ void CreateDialog::_sbox_input(const Ref<InputEvent> &p_ie) {
 	}
 }
 
+Ref<Texture> CreateDialog::_get_editor_icon(const String &p_type) const {
+
+	if (has_icon(p_type, "EditorIcons")) {
+		return get_icon(p_type, "EditorIcons");
+	}
+
+	const Map<String, Vector<EditorData::CustomType> > &p_map = EditorNode::get_editor_data().get_custom_types();
```

Comment:
maybe just to avoid iterating over the map, you can also do:
```
String parent = ClassDB::get_parent_class(p_type);
const Vector<EditorData::CustomType> &ct = EditorNode::get_editor_data().get_custom_types()[parent];
// and then iterate

```